### DISCUSSION TITLE
Getting "ERROR:  type "public.geography" does not exist"

### DISCUSSION THREAD

**Discussion by bradCypress** [supabase/supabase#4474] [Questions] [answered]
I enabled POSTGIS and POSTGIS_TOPOLOGY and tried to create a new table with a Geography column, but I am getting the above error.

From pgAdmin I do see the that both extensions are avaialable.

**Comment by GaryAustin1** [comment]
From your limited description I would look into search paths.  I use PostGIS and did not have an issue, but I have "extensions" in my search path.  The extensions are installed in extensions schema by default and the message implies something is looking for it in public schema.  This is more just a hint though.

**Comment by bradCypress** [comment]
It took a while (Supabase docs are still very thin) but I found out how to restart the server (gear icon, Restart Server button in the Infrastructure section) and that fixed it. So I assume that some extensions require a reboot.

**Reply by bradCypress** [reply]
Ok I spoke too soon. It did work *once* after restarting the server. I tried creating a second table and the old error is back. I did try it with and without the "public" schema prefix which is in my backup file. I just tried extensions.geography and that did work. back to Google I need to figure to add it to the path.

**Reply by GaryAustin1** [reply]
I set search path in SQL editor with:
set search_path = "$user", public, auth, extensions

And I have extra search path under API settings in the UI set to "public, extensions"

**Reply by bradCypress** [reply]
I am getting very random results.

alter database postgres set search_path = "$user", public, topology; worked for me, then stopped working. Adding extensions is now working. But sometimes the set search_path does not work. It returns success but 'show search_path' is emtpy.

Something in the header of my backup file is clearing the path when I try to import the database. Looks like this line: 

SELECT pg_catalog.set_config('search_path', '', false);

setting search_path fails for a while after this and then just starts working again. Serious effing weirdness! I am having trouble isolating the variables so I can properly debug this.

Back to the battle!

**Comment by bradCypress** [comment, marked as answer]
All happy now:

So if I modify my backup file to:

Remove the `SELECT pg_catalog.set_config('search_path', '', false);` line
Add `set search_path = "$user", public, auth, extensions;`
and change `public.geography` to just `geography` on all my tabledefs.
(and repeat for all 53 backup files- luckily I have it scripted) then all is working and my application seems happy with the new db location.

I am learning that scripting backup file mods to move between various Postgres platforms is a useful skill. Had to do the same to move stuff from my dev server to to Azure.
